compiler-core/src/language_server/imports.rs

```diff
@@ -0,0 +1,77 @@
+use ecow::EcoString;
+use lsp_types::{Position, Range, TextEdit};
+
+use crate::{
+    ast::{Definition, Import, TypedDefinition},
+    build::Module,
+    line_numbers::LineNumbers,
+};
+
+// Gets the position of the import statement if it's the first definition in the module.
+// If the 1st definition is not an import statement, then it returns the 1st line.
+// 2nd element in the pair is true if the first definition is an import statement.
+pub fn first_import_in_module(module: &Module, line_numbers: &LineNumbers) -> (Position, bool) {
+    // As "self.module.ast.definitions"  could be sorted, let's find the actual first definition by position.
+    let first_definition = module
+        .ast
+        .definitions
+        .iter()
+        .min_by(|a, b| a.location().start.cmp(&b.location().start));
+    let import = first_definition.and_then(get_import);
+    let import_start = import.map_or(0, |i| i.location.start);
+    let import_line = line_numbers.line_number(import_start);
+    (Position::new(import_line - 1, 0), import.is_some())
+}
```

Comment:
This function doesn't seem to do what it says. I think it should return `Option<Position>` and be called `position_of_first_definition_if_import` or something like that

Reply:
I can do that. As noted below though, I didn't write any of the code in this file, simply moved it to a more accessible place.